lib/tasks/publishing_api.rake

```diff
@@ -122,4 +122,13 @@ namespace :publishing_api do
       DataHygiene::PublishingApiHtmlAttachmentRedirector.call(args[:content_id], args[:destination], dry_run: false)
     end
   end
+
+  desc "Discard all draft about pages that have the same base_path as their WorldwideOrganisation"
+  task discard_draft_worldwide_organisation_about_pages: :environment do
+    about_pages = YAML.load_file(File.join(Rails.root, 'lib', 'tasks', 'about_pages.yml'))
```

Comment:
Rather than dumping these as a YAML then re-loading them, would it be better to put the query here?  Then the YAML file wouldn't be needed and the rake task would become easily re-usable.
```suggestion
    about_pages = Edition
   .where(document_type: "about", state: 'draft')
   .select { |edition| edition.base_path =~ /\A\/world\/organisations\/[^\/]+\z/ }
   .map { |edition| [edition.document.content_id, edition.document.locale] }
```

Reply:
That's not possible because that query has to be run in `publishing_api`.
`Edition` in `whitehall` is different from `Edition` in `publishing_api`:
```
irb(main):003:0> # This is the whitehall console
=> nil
irb(main):004:0> Edition.where(document_type: "about")
Traceback (most recent call last):
ActiveRecord::StatementInvalid (Mysql2::Error: Unknown column 'editions.document_type'...
```

Reply:
Is it possible to access publishing-api using `GdsApi::PublishingApi`, e.g. https://github.com/alphagov/whitehall/blob/147b562eb6fd8f3acddcd9bdc1fa0186c2133782/lib/tasks/publishing_api.rake#L6

Reply:
Since this doesn't modify any of the models in Whitehall, could we then move this Rake task into the publishing-api? That way it would be reusable and doesn't need the YAML file.

Alternatively, am I right in thinking that we only need to run this task once and then we can remove it? If that's the case then we may as well merge this is now, run it and then remove it afterwards.

Reply:
Yes, we just need to run it once. I could have done it in the console directly, but @elliotcm and I decided that it would be a good idea to have it visible somewhere. We can definitely merge it, run it, and delete it.